platform/pc/service.c

```diff
@@ -381,42 +382,68 @@ closure_function(1, 1, void, sync_complete,
     closure_finish();
 }
 
-static void storage_shutdown(int status, status_handler completion)
+static void storage_shutdown(int status, merge m)
 {
     if ((status == 0) || !table_find(get_environment(), sym(RADAR_KEY))) {
-        storage_sync(completion);
+        storage_sync(apply_merge(m));
     } else {
-        merge m = allocate_merge(heap_locked(&heaps), completion);
-        status_handler sh = apply_merge(m);
         klog_save(status, apply_merge(m));
         storage_sync(apply_merge(m));
```

Comment:
`storage_sync(apply_merge(m))` is now executed regardless of whether the klog is being saved, so it can be moved out of the if/else block to avoid duplicated code.

Reply:
Good point, I cleaned that up, thanks!